### DISCUSSION TITLE
Can't connect to web proxy

### DISCUSSION THREAD

**Discussion by edidudedo** [pf-robotics/kachaka-api#70] [Q&A]
After running 
`./tools/web_proxy/start_proxy_remote.sh <カチャカのIPアドレス> `
to run the web proxy, I tried to run the web application. The front-end seems to running correctly. However it seems like the robot and the web are not connecting to each other, as when I tried to check the robot version or pose, it returns unknown and NaN. 

I tried to check the web proxy by running localhost:50000, but I get this error :
`upstream connect error or disconnect/reset before headers. reset reason: connection failure`

is there a way to solve this issue?

Im running kachaka-api in my docker but because I connect my docker to my local machine. I should be able to run localhost:5173 on my local machine too (I think)

this is the web proxy :
![Screenshot from 2023-12-13 20-06-03](https://github.com/pf-robotics/kachaka-api/assets/93476480/8538060d-0b94-4a03-b337-f5bddd046ed4)

this is the web sample : 
![Screenshot from 2023-12-13 20-08-05](https://github.com/pf-robotics/kachaka-api/assets/93476480/a60ede45-2aa5-4033-9dc9-f858f862ae66)

**Comment by watanabe-pfr** [comment]
I'm sorry to be late to answer. I retested the web proxy and webapp demo in my local environment and they properly worked.

`upstream connect error or disconnect/reset before headers. reset reason: connection failure`
This message is shown when a connection between the proxy and kachaka.
Please confirm the settings are correct regarding:
* Kachaka API is enabled (you can toggle enable/disable from the smartphone app)
* Kachaka and your machine (or virtual machine) is on the same network)

To reproduce the problem, could you provide more detailed information about your environment?
* OS (It was only tested on Ubuntu 20.04, so there might be some problems in macOS or other OSes)
* Container settings for web app (Are you running the web app in a docker container? If so, the network mode of the container must be "host".)
